src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+
+(define-read-only (cost_geq (n uint))
+    (runtime u166))
+
+(define-read-only (cost_leq (n uint))
+    (runtime u166))
+
+(define-read-only (cost_le (n uint))
+    (runtime u166))
+
+(define-read-only (cost_ge (n uint))
+    (runtime u166))
+
+(define-read-only (cost_int_cast (n uint))
+    (runtime u164))
+
+(define-read-only (cost_mod (n uint))
+    (runtime u168))
+
+(define-read-only (cost_pow (n uint))
+    (runtime u170))
+
+(define-read-only (cost_sqrti (n uint))
+    (runtime u167))
+
+(define-read-only (cost_log2 (n uint))
+    (runtime u161))
+
+(define-read-only (cost_xor (n uint))
+    (runtime u167))
+
+(define-read-only (cost_not (n uint))
+    (runtime u162))
+
+(define-read-only (cost_eq (n uint))
+    (runtime (linear n u7 u172)))
+
+(define-read-only (cost_begin (n uint))
+    (runtime u202))
+
+(define-read-only (cost_hash160 (n uint))
+    (runtime (linear n u1 u201)))
+
+(define-read-only (cost_sha256 (n uint))
+    (runtime (linear n u1 u100)))
+
+(define-read-only (cost_sha512 (n uint))
+    (runtime (linear n u1 u176)))
+
+(define-read-only (cost_sha512t256 (n uint))
+    (runtime (linear n u1 u188)))
+
+(define-read-only (cost_keccak256 (n uint))
+    (runtime (linear n u1 u221)))
+
+(define-read-only (cost_secp256k1recover (n uint))
+    (runtime u14344))
+
+(define-read-only (cost_secp256k1verify (n uint))
+    (runtime u13540))
+
+(define-read-only (cost_print (n uint))
+    (runtime (linear n u3 u1413)))
+
+(define-read-only (cost_some_cons (n uint))
+    (runtime u219))
+
+(define-read-only (cost_ok_cons (n uint))
+    (runtime u219))
+
+(define-read-only (cost_err_cons (n uint))
+    (runtime u219))
+
+(define-read-only (cost_default_to (n uint))
+    (runtime u249))
+
+(define-read-only (cost_unwrap_ret (n uint))
+    (runtime u299))
+
+(define-read-only (cost_unwrap_err_or_ret (n uint))
+    (runtime u339))
+
+(define-read-only (cost_is_okay (n uint))
+    (runtime u287))
+
+(define-read-only (cost_is_none (n uint))
+    (runtime u212))
+
+(define-read-only (cost_is_err (n uint))
+    (runtime u282))
+
+(define-read-only (cost_is_some (n uint))
+    (runtime u223))
+
+(define-read-only (cost_unwrap (n uint))
+    (runtime u284))
+
+(define-read-only (cost_unwrap_err (n uint))
+    (runtime u264))
+
+(define-read-only (cost_try_ret (n uint))
+    (runtime u256))
+
+(define-read-only (cost_match (n uint))
+    (runtime u286))
+
+(define-read-only (cost_or (n uint))
+    (runtime (linear n u3 u149)))
+
+(define-read-only (cost_and (n uint))
+    (runtime (linear n u3 u149)))
+
+(define-read-only (cost_append (n uint))
+    (runtime (linear n u71 u176)))
+
+(define-read-only (cost_concat (n uint))
+    (runtime (linear n u75 u244)))
+
+(define-read-only (cost_as_max_len (n uint))
+    (runtime u475))
+
+(define-read-only (cost_contract_call (n uint))
+    (runtime u153))
+
+(define-read-only (cost_contract_of (n uint))
+    (runtime u13400))
+
+(define-read-only (cost_principal_of (n uint))
+    (runtime u39))
```

Comment:
This one is a puzzle for me.
It calculates principal derived from the public key. Computational wise it is far more complex than many other functions yet it looks like it is the cheapest one.

Reply:
Yeah -- I think this may require looking more into the structure of the benchmark. Ultimately, this isn't _that_ complex of an operation: calculating a principal is just a hash160 of the inputted public key (and checking that the public key is valid) -- but this measurement is about 10x cheaper than the hash160 operation, so we should look into it.